### ISSUE TITLE
ComboBox with setPageLength=0 can be rendered useless when quickly entering value

### ISSUE THREAD

**Issue by focbenz** [vaadin/framework#10549]
*Expected:*
In case we disable the paging option in the suggestion list of the ComboBox it is possible to "break" the Vaadin ComboBox which never shows a value again.

*Actual:*
Steps to reproduce the issue:

- Simply enter a value not in the list and quickly pressing [Enter]

![vaadin8comboboxbroken](https://user-images.githubusercontent.com/23717010/35108325-8e94ce46-fc73-11e7-906c-438a785840f7.jpg)

*Affected versions:*
Issue can be observed with latest Vaadin 8.1.7 and Vaadin 8.2.1 release

*Sample code:*
```
        String comboBoxValuesString = "First Entry,Second Entry,Third entry, Fourth entry";
        List<String> comboBoxValues = Arrays.asList((comboBoxValuesString).split(","));
        // the combobox can only be broken if any value is entered before the list of suggestions is loaded
        
	    // Create new ComboBox
        ComboBox<String> demobox = new ComboBox<>();
        demobox.setPageLength(0); // <-- this option is required to break the ComboBox
        demobox.setItems(comboBoxValues);
```

**Comment by focbenz** [comment]
*Sample Testbench code to reproduce the issue*
```
$(ComboBoxElement.class).first().sendKeys("A" + Keys.ENTER);
$(ComboBoxElement.class).first().getPopupSuggestions();
// ComboBox is broken and does not show any suggestions or accept values
```

**Comment by focbenz** [comment]
Issue can still be observed with Vaadin 8.3.3 and Vaadin 8.4.3

**Comment by Vvalter** [comment]
This is still a problem in 8.4.5

**Comment by tsuoanttila** [comment]
Seems to work as expected with `8.6-SNAPSHOT`. Can you confirm @focbenz ?

**Comment by focbenz** [comment]
Sure ... I thought we put the vaadincomboboxbreaker on GitHub ... but apparently we didn't ... so it might take me some time today because of sprint end and review.

**Comment by focbenz** [comment]
@TatuLund was correct in referencing this - with the newest 8.6-SNAPSHOT the issue is fixed.

I committed the sample Maven project here with a not-working Vaadin Version 8.4.5 here:
https://github.com/FOCONIS/vaadincomboboxbreaker

If the Vaadin Version is changed to 8.6-SNAPSHOT in the Maven POM the test passes.

Good work - happy to see this finally fixed!

We would appreciate a fix in the 8.4.x or maybe 8.5.x release waggons ... if that is possible

**Comment by tsuoanttila** [comment]
The fix is API compatible, so expect it to land in the first maintenance of 8.5

**Comment by focbenz** [comment]
Closing the issue with the information fixed in 8.5.1
